src/node/structural_equal.cc

```diff
@@ -137,6 +328,13 @@ class RemapVarSEqualHandler : public SEqualReducer::Handler {
       // Caution: entry becomes invalid when the stack changes
       auto& entry = task_stack_.back();
 
+      if (entry.force_fail) {
+        if (first_mismatch != nullptr) {
```

Comment:
Should probably call `CheckResult(false)` here, if we want to continue supporting `assert_mode`